libraries/ms-common/src/x86/can_hw.c

```diff
@@ -260,8 +270,37 @@ StatusCode can_hw_transmit(uint32_t id, bool extended, const uint8_t *data, size
     // Fifo is full
     return status_msg(STATUS_CODE_RESOURCE_EXHAUSTED, "CAN HW TX failed");
   }
-  // Unblock TX thread
-  sem_post(&s_tx_sem);
```

Comment:
I think we should still TX over vcan when loopback is enabled, just without enabling loopback on the socket itself, as long as it doesn't impact the flakiness of the tests. It's better hardware emulation that way since stm32 still txes on the CAN network in loopback mode. Do you think this is possible? If not we should put a warning by the `loopback` field in can.h saying that we don't tx over vcan while in loopback mode on x86.

Reply:
Makes sense. Writing to the socket itself is pretty fast, but I was doing some benchmarking and it seems like doing it over the thread can take up to 10 ms(!) so I just threw in a `write()` call within the short circuit logic. I also updated the RX thread to kill itself, since I think the constant `select` timeouts doesn't help determinism.

Reply:
We should still be able to RX external messages while in loopback though, since that's a thing stm32 can do. E.g. for some reason loopback is enabled on centre console on master (?? thanks arshan), and we should still be able to send messages to it, as long as it doesn't cause the tests to be flaky of course.

Reply:
That's not actually true. See https://www.st.com/resource/en/reference_manual/dm00031936-stm32f0x1stm32f0x2stm32f0x8-advanced-armbased-32bit-mcus-stmicroelectronics.pdf, page 819, section 29.5.2 Loop back mode:
> The actual value of the CANRX input pin is disregarded by the bxCAN. The transmitted messages can be monitored on the CANTX pin.

Loopback is enabled on centre_console on master because Arshan made a mistake a year ago and no one caught it: https://github.com/uw-midsun/firmware_xiv/pull/59
We were both tagged on that review so we're both guilty for that :) I fixed this on mockup, but I haven't merged mockup centre_console changes to master yet.

Reply:
Oh interesting! Seems good then.